### PR TITLE
Update development md

### PR DESCRIPTION
**Pull request by jonchenn**
- removed minikube steps
- re-format local development section
- Added sb tools steps for simple build & deploy
- update TOC
- update testing section

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+Table of Content
+<!-- vscode-markdown-toc -->
+* 1. [Project Requirements](#ProjectRequirements)
+* 2. [Code Submission Process](#CodeSubmissionProcess)
+* 3. [Local Environment Setup](#LocalEnvironmentSetup)
+* 4. [Develop and Test on a GKE cluster](#DevelopandTestonaGKEcluster)
+* 5. [Advanced Skaffold commands](#AdvancedSkaffoldcommands)
+* 6. [Debugging](#Debugging)
+* 7. [Testing](#Testing)
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:11]: The numbering for the TOC looks different when viewed in the browser

Comment (jonchenn) [reply docs/DEVELOPMENT.md]: Updated: https://github.com/GoogleCloudPlatform/core-solution-services/blob/e1f369224569cf1d6966616c2f76a78f75a8cd97/docs/DEVELOPMENT.md

**Comment by lavjain** [comment]
Please use `git mv DEVELOPMENT.md docs/DEVELOPMENT.md`

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+  ```
+  # after fixing the code…
+  git commit -a -m 'another fix'
+  git push
+  ```
+  - This will update the pull request and re-run all necessary tests automatically.
+  - If all tests passed, you will need to wait for the reviewers’ approval.
+* Once the request has been approved, the reviewer or Repo Admin will merge the pull request back to the upstream `main` branch.
+
+###  2.3. <a name='2.3.ForRepoAdminsReviewingaPullRequest'></a> 2.3. (For Repo Admins) Reviewing a Pull Request
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:96]: `2.3` repeated in the above line

File: docs/DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+###  4.2. <a name='SetupWorkloadIdentityUser'></a>Set up Workload Identity User
+
+* Run the following to set up the Kubernetes Service Account (ksa) in your namespace:
+  ```
+  export PROJECT_ID=<your-dev-project-id>
+  export SKAFFOLD_NAMESPACE=$YOUR_GITHUB_ID
+  export GSA_NAME="gke-sa"
+  export KSA_NAME="gke-sa"
+  bash ./tools/bind_ksa.sh
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:245]: `bind_ksa.sh` is not currently checked in

File: docs/DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+By default, VS Code will use the Python interpreter you've selected with the Python extensions (CMD + SHIFT + P -> __Select Interpreter__) so clicking __Debug__ and running without a configuration should work, so long as you have shifted the interpreter over and activated the "Common" VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
+
+####  6.1.2. <a name='Debuggingcomponentscomponent_name'></a>Debugging `components/<component_name>`
+This should also just work, so long as you have selected the right interpreter, are in the microservice folder, and have entered your VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
+
+###  6.2. <a name='DebuggingwithSkaffoldCloudCodeonGKEcluster'></a>5.2. Debugging with Skaffold + Cloud Code (on GKE cluster)
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:340]: Typo: Extra `5.2` here

File: docs/DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+
+####  6.1.1. <a name='Debuggingcomponentscommon'></a>Debugging `components/common`
+By default, VS Code will use the Python interpreter you've selected with the Python extensions (CMD + SHIFT + P -> __Select Interpreter__) so clicking __Debug__ and running without a configuration should work, so long as you have shifted the interpreter over and activated the "Common" VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
+
+####  6.1.2. <a name='Debuggingcomponentscomponent_name'></a>Debugging `components/<component_name>`
+This should also just work, so long as you have selected the right interpreter, are in the microservice folder, and have entered your VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:338]: Sentence repeated from previous sub-section

File: docs/DEVELOPMENT.md [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+  "python.linting.enabled": true,
+  "python.linting.pylintPath": "pylint",
+  "editor.formatOnSave": true,
+  "python.formatting.provider": "yapf",
+  "python.formatting.yapfArgs": [
+    "--style={based_on_style: pep8, indent_width: 2}"
+  ],
+  "python.linting.pylintEnabled": true,
+  "terminal.integrated.env.osx": {
+    "PYTHONPATH": "${workspaceFolder}/common/src/"
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:119]: Just realized, this should be `${workspaceFolder}/components/common/src/`

File: docs/DEVELOPMENT.md [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+  },
+  "python.analysis.extraPaths": [
+    "${workspaceFolder}/common/src/"
+  ],
+  "python.autoComplete.extraPaths": [
+    "${workspaceFolder}/common/src/"
+  ]
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:126]: same for lines `122` and `125`

File: docs/DEVELOPMENT.md [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+  ```
+
+###  4.2. <a name='SetupWorkloadIdentityUser'></a>Set up Workload Identity User
+
+* Run the following to set up the Kubernetes Service Account (ksa) in your namespace:
+  ```
+  export PROJECT_ID=<your-dev-project-id>
+  export SKAFFOLD_NAMESPACE=$YOUR_GITHUB_ID
+  export GSA_NAME="gke-sa"
+  export KSA_NAME="gke-sa"
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:244]: Should this be `ksa`

File: docs/DEVELOPMENT.md [hunk 9]
```diff
@@ -0,0 +1,402 @@
...
+
+  # Go to a specific microservice folder:
+  cd components/<component_name>
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r requirements.txt
+  pip install -r requirements-test.txt
+
+  # If this component depends on the common folder:
+  pip install -r ../common/requirements.txt
```
Comment (lavjain) [inline docs/DEVELOPMENT.md:375]: Requirements.txt for `common` should preferably be installed before the `microservice`

File: docs/DEVELOPMENT.md (hunk 8 above)
Comment (jonchenn) [reply docs/DEVELOPMENT.md:244]: This has to be `gke-sa`, this is defined in the terraform 3-gke stage.

File: docs/DEVELOPMENT.md (hunk 9 above)
Comment (jonchenn) [reply docs/DEVELOPMENT.md:375]: Not in the local development. Each microservice builds image from the base_image, which is the common image. However, for local dev, you'll need to manually install those deps.

**Review by lavjain — APPROVED** [review]
LGTM

**Pull request merged**